File: 2021/09/15/main.go
package main

func maxTurbulenceSize(arr []int) int {
	g, l, r := 1, 1, 1
	for i := 1; i < len(arr); i++ {
		switch {
		case arr[i] > arr[i-1]:
			g, l = l+1, 1
			r = max(r, g)
		case arr[i] < arr[i-1]:
			g, l = 1, g+1
			r = max(r, l)
		default:
			g, l = 1, 1
		}
	}

	return r
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: 2021/09/15/main.py
class Solution:
    def maxTurbulenceSize(self, arr: List[int]) -> int:
        g, l, r = 1, 1, 1
        
        for i in range(1, len(arr)):
            if arr[i] > arr[i-1]:
                g, l = l + 1, 1
                r = max(r, g)
            elif arr[i] < arr[i-1]:
                g, l = 1, g+1
                r = max(r, l)
            else:
                g, l = 1, 1
        
        return r